botocore/utils.py

```diff
@@ -374,15 +374,15 @@ def _fetch_metadata_token(self):
                     raise BadIMDSRequestError(request)
             except ReadTimeoutError:
                 return None
+            except RETRYABLE_HTTP_ERRORS as e:
```

Comment:
It doesn't look like we have a test that actually exercises this code path. Can we get something mocked up in the functional tests that validates this behavior is working correctly?